### ISSUE TITLE
Gratuitous warnings for x509 when reading from stdin

### ISSUE THREAD

**Issue by dsnyder0pc** [openssl/openssl#22893]
`openssl x509` should not print warnings by default when reading from stdin:

https://github.com/openssl/openssl/blob/master/apps/x509.c#L704-L706
https://github.com/openssl/openssl/blob/master/apps/x509.c#L757-L759

For example, I frequently fetch certs from HashiCorp vault and use `openssl x509` to check expiry dates:

```
$ vault kv get -field=cert /path/to/cert | openssl x509 -noout -dates
```

Previously, I would get output like this:
```
notBefore=Dec  8 18:26:59 2022 GMT
notAfter=Dec  8 18:26:59 2023 GMT
```

Now, I get a warning that I have no way to suppress besides redirecting all error messages from openssl to `/dev/null`:
```
Warning: Reading certificate from stdin since no -in or -new option is given
notBefore=Dec  8 18:26:59 2022 GMT
notAfter=Dec  8 18:26:59 2023 GMT
```

If `-in` is now mandatory, the code should be rewritten to accept `-in -` as that's a common Unix convention for intentionally reading from stdin.

**Comment by t8m** [comment]
**OTC: Do we want to revert these warnings?**

As far as I remember they were added because there were bug reports that running `openssl x509` is stuck without any message when it was actually reading from stdin.

**Comment by dsnyder0pc** [comment]
Every UNIX command that reads from stdin by default appears to be stuck when stdin is a terminal. Users should expect this. :)

If you still want to keep the warning message, I recommend calling `isatty()` on stdin first. Only output the warning message if stdin is mapped to a terminal. This gives the desired behavior while not emitting warning messages for users who know what they are doing.

Adding `-in -` for intentional syntax would be nice. As would offering a `-quiet` option to suppress warning messages. However, the `isatty()` call is a solution that follows the _principle of least astonishment._

**Comment by t8m** [comment]
Thank for the idea to use isatty() to check. I think this is what we should do.

**Comment by t8m** [comment]
**OTC: Using isatty() to suppress the warning seems reasonable.**

**Comment by na4ma4** [comment]
A workaround (for linux/macOS) if this is breaking things for you is `openssl x509 -in /dev/stdin`.

**Comment by bkane-msft** [comment]
thanks @na4ma4 , I was able to [update my easyssl](https://github.com/bbkane/dotfiles/commit/5a3c23014e5d1a251b488eccbb8c7c317737a34e) script and silence the warnings!

**Comment by latchee** [comment]
> A workaround (for linux/macOS) if this is breaking things for you is `openssl x509 -in /dev/stdin`.

Would anybody now how to apply similar workaround in the MS Windows world? Indeed, /dev/stdin is not available there and this warning pops up.

**Comment by x-yuri** [comment]
Introduced in [`3.2.0`][b], fixed in [`3.3.0`][a] AFAICS.

[a]: https://github.com/openssl/openssl/commit/5c846d32d4a1dc7ee7934bc867b9941809b76beb
[b]: https://github.com/openssl/openssl/commit/611ef4f3737cc5812bdefe381403fdf1bacfba06#diff-0fb19438117c9272e0d4a812eab3dfd079ad93e5e598fda46a6f2eb65fa9ad16R733

**Comment by t8m** [comment]
It will be fixed in 3.2.2 as well.

**Comment by boris-kolpackov** [comment]
> It will be fixed in 3.2.2 as well.

FYI, we just tried to narrow the version range for this workaround to 3.2.0-3.2.2 but were bitten by `openssl` on Debian which say it's `3.2.2-dev` but doesn't include this fix.

**Comment by dsnyder0pc** [comment]
Thanks all for your work on this! The fix made it to Homebrew some weeks ago:
```
$ brew info openssl
==> openssl@3: stable 3.3.1 (bottled)
```
